zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ColumnFamilyContext.java

```diff
@@ -113,4 +114,13 @@ public void withPrefixKey(final DbKey key, final ObjIntConsumer<byte[]> prefixKe
       prefixKeyBuffers.add(prefixKeyBuffer);
     }
   }
+
+  ByteBuffer keyWithColumnFamily(DbKey key) {
+    final var bytes = ByteBuffer.allocate(Long.BYTES + key.getLength());
+    final var buffer = new UnsafeBuffer(bytes);
+
+    buffer.putLong(0, columnFamilyPrefix, ZeebeDbConstants.ZB_DB_BYTE_ORDER);
+    key.write(buffer, Long.BYTES);
+    return bytes;
+  }
```

Comment:
Previously we just used the `prefixKey` provided by `ColumnFamilyContext#withPrefixKey`. The `prefixKey` was constructed like this: https://github.com/camunda/zeebe/blob/f227c3201ef14764dd2c000e5cf2d229482f76d6/zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ColumnFamilyContext.java#L108-L109

Now we don't want to seek to the `prefixKey` anymore. Instead we want to seek to `startAt`. But `startAt` is just a arbitrary `DbKey`. We need to tell RocksDB about the logical column family where we expect this key to be in. That's why we write the logical column family id followed by the `startAt` key.

So semantically there's no change, we pretty much the same thing that was done before. We just have to do it explicitly now because we can't use `ColumnFamilyContext#withPrefixKey` for this.